### ISSUE TITLE
Refinements do not work inside of a `class_exec`

### ISSUE THREAD

**Issue by alecgorge** [oracle/truffleruby#1949]
### Truffleruby Version

`truffleruby 20.0.0, like ruby 2.6.5, GraalVM CE Native [x86_64-darwin]`

### Code to reproduce

```ruby
module Refinements
  refine String do
    def test_refinement
      'it works!'
    end
  end
end

class TestClass

end

module Test
  using Refinements

  def self.test
    puts 'cars'.test_refinement
    TestClass.class_exec(&method(:test_internal))
  end

  def self.test_internal
    # this is needed to make this work
    # using Refinements
    puts 'cars'.test_refinement
  end

  # also doesn't work
  # def self.test2
  #   puts 'cars'.test_refinement
  #   TestClass.class_exec do
  #     puts 'cars'.test_refinement
  #   end
  # end
end

Test.test
```

### Expected output (MRI)

```
it works!
it works!
```

### Actual Output

```
it works!
/Users/alecgorge/Library/Preferences/RubyMine2019.3/scratches/scratch_5.rb:24:in `test_internal': undefined method `test_refinement' for "cars":String (NoMethodError)
	from /Users/alecgorge/Library/Preferences/RubyMine2019.3/scratches/scratch_5.rb:21:in `test_internal'
	from /Users/alecgorge/Library/Preferences/RubyMine2019.3/scratches/scratch_5.rb:18:in `class_exec'
	from /Users/alecgorge/Library/Preferences/RubyMine2019.3/scratches/scratch_5.rb:18:in `test'
	from /Users/alecgorge/Library/Preferences/RubyMine2019.3/scratches/scratch_5.rb:28:in `<main>'
```

**Comment by alecgorge** [comment]
FYI, my working around of putting `using Refinements` in the method works on Truffleruby but when run on MRI I get:

```
Module#using is not permitted in methods
```

**Comment by norswap** [comment]
Could be fixed by https://github.com/oracle/truffleruby/pull/1988

**Comment by bjfish** [comment]
This appears fixed now on the latest master: aa9d75a31be7fa58b1a4c0ed5173670229f35082. 
I assume it was fixed by #1988 .
```
truffleruby 20.2.0-dev-aa9d75a3, like ruby 2.6.5, Interpreted JVM [x86_64-darwin]
it works!
it works!
```
